Fix cyclic concurrent buildings of vite and hugo while both in watching mode

Convolutio opened the pull request:
When `npm start` is run, `vite build` is watching wrong directories in which files are edited by the final `hugo` building 
server, launching a new build from `vite`, then a new build from `hugo`, and so on. This is unusable for an edition of the layout files or the style sheets with real-time rendering because the website is uselessly reloaded too much frequently.

With this fix, `vite` rebuilds the website only if useful files are edited, i.e. files in the `app`, `layout` or `style` directories.

matthewruzzi reviewed (changes requested):
This pr contains many unrelated changes.

Convolutio commented:
You are right, I do not know why I have tried to merge such a long branch. I will propose a more relevant branch soon.

Convolutio committed:
fix: permanent circular build in concurrent watch

also only try to remove the static/svelte/index.html file

commit 0386e30a5b24eaffea1a576de3d0ddb523b850c3
Author: Convolutio <79534557+Convolutio@users.noreply.github.com>
Date:   Sun Oct 12 13:06:01 2025 +0200

    fix: json indentation

commit 714ecf5c7acf84f5194102bf0da8716033242131
Author: Convolutio <79534557+Convolutio@users.noreply.github.com>
Date:   Sun Oct 12 12:59:34 2025 +0200

    fix: remove the useless log

commit 856d92723da0e0c4f5a4803e4f5a86d024466871
Author: Convolutio <79534557+Convolutio@users.noreply.github.com>
Date:   Sat Jun 15 21:46:21 2024 +0200

    fix: circular permanent build in concurrent watch

Convolutio commented:
Now there is only the proposed fix in this PR (sorry for the confusion).

To test it, just run `npm start`, edit a `scss` file in the `styles` directory, and check if the concurrent rebuild is not permanently looping.

NB: I have tested with Hugo `v0.122.0`. With last versions of Hugo, the hugo build fails in cause of a deprecated usage of the tweet shortcode. Another PR should be accepted with a fix of this usage for a migration with the last version (or a fix of the readme if the maintainers want to freeze the versions of hugo for this project).
